modules/core/04-channel/keeper/upgrade.go

```diff
@@ -355,11 +355,19 @@ func (k Keeper) ChanUpgradeOpen(
 		if !found {
 			return errorsmod.Wrapf(types.ErrUpgradeNotFound, "failed to retrieve channel upgrade: port ID (%s) channel ID (%s)", portID, channelID)
 		}
+		// If counterparty has reached OPEN, we must use the upgraded connection to verify the counterparty channel
```

Comment:
linky-link more, opened issue on spec repo to keep track https://github.com/cosmos/ibc/issues/1004